### ISSUE TITLE
UK: Problem adding new election on the embedded “Add new candidate” form

### ISSUE THREAD

**Issue by andylolz** [mysociety/yournextrepresentative#773]
1. go to [the London mayoral page](https://candidates.democracyclub.org.uk/election/mayor.london.2016-05-05/post/mayor:london/mayor-of-london)
2. Click “Add a new candidate”
3. Click “Add a new election for this person”
4. Pick one
5. …
6. :boom::boom::boom:

Looks like the problem is with `window.location.pathname` just here:
https://github.com/DemocracyClub/yournextrepresentative/blob/0cfb220e/candidates/static/js/person_form.js#L158

**Comment by jamesvasey** [comment]
This appears to be a poor interface / user issue. - The 'add a new election' isn't (in most cases) necessary based on the route you have accessed.

If you go to the London Mayoral Page and add a candidate, the person you are adding will automatically be added to the 'London Mayoral' election, without the need to 'Add a new election for this person'. 

If they have stood previously for any general election (or other in the database) find the candidate, and add the election that way. 

But if you going via the election route to add them, the 'add a new election' is unnecessary, - except if they are standing in multiple elections. In which case the work around is to add the candidate first for the mayoral election, and then go back in add them for another (assembly / borough) election. - 
It isn't perfect, but it gets around the problem.

**Comment by andylolz** [comment]
@jamesvasey: I totally agree – I think the workaround you suggest is perfectly fine!

I think there is still a bug to fix here, though. I’d suggest removing the “Add a new election for this person” from this form, to prevent weirdness from happening. The alternatively would be to fix it so it does the right thing… But I’m not sure it will be that useful, and is probably harder to do!

**Comment by jamesvasey** [comment]
I think the 'add candidate form' is generic rather than being redesigned (or re-rendered) based on which page the user has come from - which would fix the problem. I agree it needs resolving, I just wanted offer the workaround as I didn't want anyone to be stuck in a loop.

**Comment by andylolz** [comment]
> I think the 'add candidate form' is generic rather than being redesigned (or re-rendered) based on which page the user has come from

That’s right, yes. [I believe it’s this template](https://github.com/DemocracyClub/yournextrepresentative/blob/0cfb220e/elections/uk/templates/candidates/_person_form.html). But I suspect it can be tweaked depending on which page it’s displayed on. I’ll have a go at fixing this.
